Crash when editing a .org file

### Summary

Helix crashes consistently when an .org file contains a * followed by a blank line.

I understand that Helix does not support .org files, but I would expect them to be treated as text files.
Note that the problem does now occur when the file extension is changed to .txt.

### Reproduction Steps

Two ways to reproduce this, first interactively:
> touch test.org
> hx test.org

Enter insert mode and insert a single *.
Return to normal mode.
Use "o" to open a new line below the current line.

The editor freezes and, after a short wait, crashes.

Alternatively:
> echo "*\n" > test.org
> hx test.org

The above results is a crash.

### Helix log

2022-08-09T17:03:52.486 helix_view::theme [WARN] Theme: malformed hexcode: grey09
2022-08-09T17:03:52.488 helix_view::theme [WARN] Theme: malformed hexcode: grey08
2022-08-09T17:03:52.488 helix_view::theme [WARN] Theme: malformed hexcode: grey0D


### Platform

Linux (Arch)

### Terminal Emulator

Alacritty

### Helix Version

helix 22.05 (404defb5)

Replicated the issue. When attempting to delete the trailing `\r\n`, Helix just freezes with no additional logs when using `-vvv`.

**Version**: helix 22.05 (e6d6d882)

```
2022-08-09T18:52:38.511 helix_view::editor [ERROR] Failed to initialize the LSP for `source.org` { LSP not defined }
2022-08-09T18:52:44.307 helix_loader [DEBUG] Located configuration folders: []
2022-08-09T18:52:44.321 helix_view::editor [ERROR] Failed to initialize the LSP for `source.org` { LSP not defined }
2022-08-09T18:52:52.622 helix_term::commands [TRACE] entering insert mode with sel: Selection { ranges: [Range { anchor: 3, head: 5, horiz: Some(0) }], primary_index: 0 }, text: "*\r\n\r\n"
```

--- a/languages.toml
+++ b/languages.toml
@@ -1157,7 +1157,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "org"
-source = { git = "https://github.com/milisims/tree-sitter-org", rev = "1c3eb533a9cf6800067357b59e03ac3f91fc3a54" }
+source = { git = "https://github.com/milisims/tree-sitter-org", rev = "698bb1a34331e68f83fc24bdd1b6f97016bb30de" }
 
 [[language]]
 name = "solidity"

